icu4c/source/i18n/unicode/calendar.h

```diff
@@ -1143,26 +1133,6 @@ class U_I18N_API Calendar : public UObject {
      */
     virtual int32_t getActualMinimum(UCalendarDateFields field, UErrorCode& status) const;
 
-#ifndef U_HIDE_DEPRECATED_API
```

Comment:
We say that we don't remove deprecated APIs. Removing ones that we never implemented is fine, because no one can ever have used them, but removing ones that do have implementations is problematic.

https://unicode-org.github.io/icu/userguide/icu/design.html#icu-api-compatibility
“Use of deprecated APIs is strongly discouraged, but they are retained for backward compatibility.”

Reply:
I think this is what @pedberg-icu had brought up during the internal review. Though these are functions that could be implicitly castable from `EDateFields` to `UCalendarDateFields`. I would not consider these implemented since C++ mangling stores the complete parameter typename. The symbols ultimately don't match up.
e.g. 
```
$ c++filt __ZN3icu8Calendar5clearENS0_11EDateFieldsE
icu::Calendar::clear(icu::Calendar::EDateFields)
```

If we want to keep these declarations to avoid potentially disrupting client code, I agree adding explicit definitions that cast and call the non-deprecated `UCalendarDateFields` versions is a good approach. What do you think?

Reply:
This was discussed in this weeks ICU-TC meeting and we are removing these function decls.